apps/hash-graph/lib/graph/src/store/test_graph.rs

```diff
@@ -0,0 +1,146 @@
+use serde::{Deserialize, Serialize};
+use time::OffsetDateTime;
+use type_system::{url::VersionedUrl, DataType, EntityType, PropertyType};
+
+use crate::{
+    identifier::{
+        knowledge::{EntityRecordId, EntityTemporalMetadata},
+        ontology::OntologyTypeRecordId,
+        time::{LeftClosedTemporalInterval, TransactionTime},
+    },
+    knowledge::{Entity, EntityProperties, LinkData},
+    ontology::OntologyType,
+    provenance::{OwnedById, ProvenanceMetadata},
+};
+
+#[derive(Debug, Default, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CustomEntityMetadata {
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub provenance: Option<ProvenanceMetadata>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub archived: Option<bool>,
+}
+
+impl CustomEntityMetadata {
+    #[must_use]
+    const fn is_empty(&self) -> bool {
+        self.provenance.is_none() && self.archived.is_none()
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EntityMetadata {
```

Comment:
Guessing there isn't a way for us to easily reuse the existing `EntityMetadata` struct?

Reply:
Nope, sadly this is not possible. But maybe we should always use this struct everywhere and bring the HASH backend closer to the BP spec (not in this PR but as a potential follow-up in the near future)

Reply:
> But maybe we should always use this struct everywhere and bring the HASH backend closer to the BP spec

That is more aligned with what I was thinking, it might be nice to actually explicitly label our extensions as being extensions / custom